streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/AssignmentInfo.java

```diff
@@ -328,13 +344,21 @@ public static AssignmentInfo decode(final ByteBuffer data) {
                     assignmentInfo.errCode = in.readInt();
                     break;
                 case 6:
+                    commonlySupportedVersion = in.readInt();
+                    assignmentInfo = new AssignmentInfo(usedVersion, commonlySupportedVersion);
+                    decodeActiveTasks(assignmentInfo, in);
+                    decodeStandbyTasks(assignmentInfo, in);
+                    decodeActiveAndStandbyHostPartitions(assignmentInfo, in);
+                    assignmentInfo.errCode = in.readInt();
+                    break;
                 case 7:
                     commonlySupportedVersion = in.readInt();
                     assignmentInfo = new AssignmentInfo(usedVersion, commonlySupportedVersion);
                     decodeActiveTasks(assignmentInfo, in);
                     decodeStandbyTasks(assignmentInfo, in);
                     decodeActiveAndStandbyHostPartitions(assignmentInfo, in);
                     assignmentInfo.errCode = in.readInt();
+                    assignmentInfo.nextRebalanceMs = in.readLong();
```

Comment:
I'm wondering if we can avoid encoding this for the common case where there is no "next rebalance time". Do you have thoughts on the subject?

Reply:
I think it's only worth doing if we don't intend to switch over to generated protocol code for the `AssignmnentInfo` anytime soon. I'm pretty sure it's not possible to accomplish this in the json schema, so it would be a short lived optimization and/or require us to bump the protocol version unnecessarily once we do switch over.

Reply:
Of course, we don't _have_ to move the `AssignmentInfo` to generated code as well (although imo it's a bit weird to do it for the subscription but not the assignment). Or we could probably just add support for this ourselves at some later point.

Do you think we should try and save these 8 bytes per assignment?

Reply:
Yeah, this is a good reason to hold off. We can always bump the version later to try a different encoding.